src/main/java/org/wildfly/security/auth/client/MatchHostRule.java

```diff
@@ -31,23 +33,46 @@
 class MatchHostRule extends MatchRule {
 
     private static final Pattern validHostSpecPattern = Pattern.compile("(?i:[-a-z0-9_]+(?:\\.[-a-z0-9_]+)*\\.?)");
+    static final Pattern validHostSpecIPv6Pattern = Pattern.compile("(?i:(?:\\:\\:)?([a-f0-9]{1,4}(?:\\:\\:?[a-f0-9]{1,4})*" +
```

Comment:
This is not great, as it accepts a wide variety of strings which are not valid IPv6 addresses, but for now it's OK.